src/main/java/dp/xlsx/DatasetFormatter.java

```diff
@@ -160,16 +167,32 @@ private void addHeaderRow(Collection<String> timeLabels) {
 
     private void addTimeLabelCells(Collection<String> timeLabels, Row headerRow, int columnOffset) {
 
+        final String[] additionalHeaders = file.getAdditionalHeaders();
+
         // write time labels across the title row
         for (String timeLabel : timeLabels) {
             Cell cell = headerRow.createCell(columnOffset);
-            cell.setCellStyle(valueRightAlign);
+            cell.setCellStyle(workBookStyles.getValueRightAlignStyle());
             cell.setCellValue(timeLabel);
             columnOffset++;
+
+            for (String additionalHeader : additionalHeaders) {
+                addHeaderCell(columnOffset, headerRow, additionalHeader + " (" + timeLabel + ")");
+                columnOffset++;
+            }
         }
     }
 
 
+    private void setCellValueByType(Cell cell, String value) {
+
+        if (value.chars().allMatch(Character::isDigit)) {
```

Comment:
will have to check this doesn't add too much overhead compared to the previous implementation